rocAL/rocAL_pybind/examples/rocAL_api_coco_pipeline_segmentation.py

```diff
@@ -0,0 +1,531 @@
+            PIL_image = cv2.cvtColor(PIL_image, cv2.COLOR_RGB2BGR)
+            for box in self.target_batch[i].bbox:
+                x1, y1, x2, y2 = box.cpu().numpy().astype(np.int)
+                cv2.rectangle(PIL_image, (x1,y1), (x2,y2), (255, 0, 0), 2)
+            cv2.imwrite(f'{img_name}.jpg', PIL_image)
+        return self.img_list_obj, self.target_batch
+
+
+    def reset(self):
+        self.loader.rocalResetLoaders()
+
+    def __iter__(self):
+        self.loader.rocalResetLoaders()
+        return self
+
+def main():
+    if len(sys.argv) < 5:
+        print('Please pass the folder image_folder Annotation_file cpu/gpu batch_size')
+        exit(0)
+
+    image_path = sys.argv[1]
+    ann_path = sys.argv[2]
+    if(sys.argv[3] == "cpu"):
+        _rali_cpu = True
+    else:
+        _rali_cpu = False
+    bs = int(sys.argv[4])
+    nt = 1
+    di = 0
+    random_seed = random.SystemRandom().randint(0, 2**32 - 1)
+
+    pipe = Pipeline(batch_size=bs, num_threads=nt,device_id=di, seed=random_seed, rocal_cpu=_rali_cpu)
+
+    with pipe:
+        jpegs, bboxes, labels = fn.readers.coco(
+            file_root=image_path, annotations_file=ann_path, random_shuffle=True, seed=di, masks=True)
+        images_decoded = fn.decoders.image(jpegs, output_type=types.RGB, file_root=image_path, annotations_file=ann_path, shard_id=di, num_shards=nt, random_shuffle=True, seed=di)
+        coin_flip = fn.random.coin_flip(probability=0.5)
+        rmn_images = fn.resize_mirror_normalize(images_decoded,
+                                            device="gpu",
+                                            output_dtype=types.FLOAT16,
+                                            output_layout=types.NCHW,
+                                            resize_min = 1344,
+                                            resize_max = 1344,
+                                            mirror=coin_flip,
+                                            mean= [102.9801, 115.9465, 122.7717],
+                                            std = [1. , 1., 1.])
+        pipe.set_outputs(rmn_images)
+
+    pipe.build()
+    if(_rali_cpu):
+        data_loader = ROCALCOCOIterator(
+            pipe, multiplier=pipe._multiplier, offset=pipe._offset, device="cpu")
+    else:
+        data_loader = ROCALCOCOIterator(
+            pipe, multiplier=pipe._multiplier, offset=pipe._offset, device="cuda")
+    epochs = 1
+    for epoch in range(int(epochs)):
+        print("EPOCH:::::",epoch)
+        for i, (images, targets) in enumerate(tqdm(data_loader)):
+            print("*******************************",i,"************************")
+            print("****************IMAGES****************")
+            print(images.image_sizes)
+            print("\nBBOXES:\n", targets[0].bbox)
+            print("\nLABELS:\n", targets[0].extra_fields['labels'])
+            print("\nMASK POLYGONS :\n", targets[0].extra_fields['masks'])
+        data_loader.reset()
+
```

Comment:
This example file does not seem to comply with other examples present . Can we make it uniform by passing args like the other example files ?